### PR TITLE
Fix #274: Added drag-and-drop feature for roadmap view of stories

### LINKED ISSUE #274: [Feature] Add the feature to drag and drop stories
## Is your feature request related to a problem? Please describe.

Currently, the status of stories can be updated only by the admin. This makes it difficult for the manager or other permitted-members to update the status of a story.

## Describe the solution you'd like

There should be a drag and drop facility to update the story, that is accessible only to the permitted members.

## Additional context

NA

### PR DESCRIPTION
**Pull request by kailash360**
**Issue Number**

fixes #274 

**Describe the changes you've made**

I have added `react-beautiful-dnd` to add the drag and drop feature. Each droppable region is now wrapped in a `<Droppable>` tag and a story card, which can be dragged, is now wrapped in a `<Draggable>` tag. I have also integrated the permissions such that the users who have the permission to `Edit Story` or `Update Story Status` will be allowed to use this feature. I have added the same in the documentation.

**Describe if there is any unusual behavior (Any Warning) of your code(Write `NA` if there isn't)**

NA

**Additional context (OPTIONAL)**

NA

**Checklist**

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] All new and existing tests passed.
- [x] The title of my pull request is a short description of the requested changes.

**Provide a Deployed link of route/page that needs to review**

https://user-images.githubusercontent.com/75155230/194617709-3f7c4a6b-81e5-42c0-89b1-e63ad7caf07f.mp4

### CODE REVIEW

File: src/components/StatusContainer.jsx [hunk 1]
```diff
@@ -1,77 +1,122 @@
-import React from 'react'
+import React, { useEffect } from 'react'
 import { navigate, Link } from '@reach/router'
 import { EOS_COMMENT, EOS_ATTACHMENT } from 'eos-icons-react'
 import { strip } from '../utils/filterText'
+import { Draggable, Droppable } from 'react-beautiful-dnd'
 import Vote from './Vote'
 
+function StoryCard(props) {
```
Comment (cyntss) [inline src/components/StatusContainer.jsx:8]: Let's move the StoryCard component to its own file

Comment (cyntss) [reply src/components/StatusContainer.jsx:8]: Always go for an isolation of components approach. Remember that the better controlled the environment of your applications the more control you will have to scale, debug, and test components and modules.

File: src/components/Stories.js [hunk 2]
```diff
@@ -145,6 +149,60 @@ const Stories = ({ authorId, followerId }) => {
...
+    const getPermissions = async () => {
+      if (!userId) return
+
+      const permissionResponse = await userStory.getPermissionsById(userId)
+      const permissions =
+        permissionResponse.data.data.user.access_role[0].permissions.map(
+          (item) => item.name
+        )
+
+      const updatedAllowed = permissions.includes('Update Story Status')
```
Comment (cyntss) [inline src/components/Stories.js:162]: Do we have this role included in the new documentation? I think I only saw for Edit story.

**Review by cyntss — CHANGES_REQUESTED** [review]

File: src/components/StatusContainer.jsx [hunk 3]
```diff
@@ -1,77 +1,122 @@
...
+import { Draggable, Droppable } from 'react-beautiful-dnd'
 import Vote from './Vote'
 
+function StoryCard(props) {
+  const { story, index, isDragAllowed } = props
+  return (
+    <Draggable
+      draggableId={story.id}
+      index={index}
+      isDragDisabled={!isDragAllowed}
```
[the remark this answers is not in the document]
Comment (kailash360) [reply src/components/StatusContainer.jsx]: Alright

File: src/components/StatusContainer.jsx (hunk 1 above)
Comment (kailash360) [reply src/components/StatusContainer.jsx]: Ok

File: src/components/Stories.js (hunk 2 above)
Comment (kailash360) [reply src/components/Stories.js:162]: Yes, I have added it

The third point under backend config contains it.
https://github.com/EOS-uiux-Solutions/user-story/blob/drag-drop-274/README.md#backend-configurations

**Comment by kailash360** [comment]
@cyntss 
I have pushed the changes

File: src/components/Stories.js (hunk 2 above)
Comment (cyntss) [reply src/components/Stories.js:162]: oh I missed it. thank you!

**Review by cyntss — APPROVED** [review]

**Pull request merged**